harambe/parser/parser.py

```diff
@@ -50,6 +56,12 @@ def validate(self, data: dict[str, Any], base_url: URL) -> dict[str, Any]:
         self.field_types = self._get_field_types(base_url)
         self.model = self._schema_to_pydantic_model(self.schema)
         cleaned_data = trim_keys_and_strip_values(data)
+        all_required_fields = self._get_all_required_fields(self.schema)
+        missing_fields = self._find_missing_required_fields(
+            cleaned_data, all_required_fields
+        )
```

Comment:
We should think about maybe implementing the visitor pattern down the road so we dont need to iterate recursively for each new check